pkg/resource/resource.go

```diff
@@ -280,41 +295,47 @@ func CRUD(obj *unstructured.Unstructured, releaseInstalled bool, owner v1.Object
 	err := clients.Interface.Get(context.TODO(), key, found)
 
 	if apierrors.IsNotFound(err) {
+		oneTimer, err := IsOneTimer(obj)
+		if err != nil {
+			return fmt.Errorf("could not determine if the object is a one-timer: %w", err)
+		}
 		// We are not recreating all objects if a release is already installed
-		if releaseInstalled && IsOneTimer(obj) {
+		if releaseInstalled && oneTimer {
 			logg.Info("Skipping creation")
 			return nil
 		}
 
 		logg.Info("Not found, creating")
 
 		logg.Info("Release", "Installed", releaseInstalled)
-		logg.Info("Is", "OneTimer", IsOneTimer(obj))
+		logg.Info("Is", "OneTimer", oneTimer)
 
 		hash.Annotate(obj)
 
 		// If we create the resource set the owner reference
-		err := controllerutil.SetControllerReference(owner, obj, RuntimeScheme)
-		exit.OnError(err)
+		if err = controllerutil.SetControllerReference(owner, obj, RuntimeScheme); err != nil {
+			return fmt.Errorf("could not set the owner reference: %w", err)
+		}
 
 		SetMetaData(obj, name, namespace)
 
-		if err := clients.Interface.Create(context.TODO(), obj); err != nil {
+		if err = clients.Interface.Create(context.TODO(), obj); err != nil {
 			if apierrors.IsForbidden(err) {
-				return errors.Wrap(err, "API error is forbidden")
+				return fmt.Errorf("API error: forbidden [%w]", err)
```

Comment:
Maybe use the same format as other wrapped errors? `wrapper text: original error`

Reply:
Right, back to the regular format.